presto-orc/src/main/java/com/facebook/presto/orc/writer/StreamOrderingLayout.java

```diff
@@ -0,0 +1,219 @@
+import com.facebook.presto.orc.proto.DwrfProto;
+import com.facebook.presto.orc.stream.StreamDataOutput;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class StreamOrderingLayout
+        implements StreamLayout
+{
+    private final DwrfStreamOrderingConfig config;
+    private final StreamLayout nonStreamOrderingLayout;
+
+    public StreamOrderingLayout(
+            DwrfStreamOrderingConfig config,
+            StreamLayout layout)
+    {
+        this.config = requireNonNull(config);
+        this.nonStreamOrderingLayout = requireNonNull(layout);
+    }
+
+    private static class StreamMetadata
+    {
+        // <Column Id, Sequence ID> -> List<Streams>>
+        private final Map<ColIdSequenceId, List<StreamDataOutput>> seqIdToStreams;
+        // <Column Id, KeyId> -> SequenceId
+        private final Map<ColIdKeyId, Integer> keyIdToSeqId;
+
+        public StreamMetadata(Map<ColIdSequenceId, List<StreamDataOutput>> seqIdToStreams, Map<ColIdKeyId, Integer> keyIdToSeqId)
+        {
+            this.seqIdToStreams = requireNonNull(seqIdToStreams, "seqIdToStreams cannot be null");
+            this.keyIdToSeqId = requireNonNull(keyIdToSeqId, "keyIdToSeqId cannot be null");
+        }
+    }
+
+    private static class ColIdKeyId
+    {
+        private final int columnId;
+        private final DwrfProto.KeyInfo keyInfo;
+
+        public ColIdKeyId(int inputColumnId, DwrfProto.KeyInfo inputKeyId)
+        {
+            this.columnId = inputColumnId;
+            this.keyInfo = requireNonNull(inputKeyId, "inputKeyId cannot be null");
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColIdKeyId)) {
+                return false;
+            }
+            ColIdKeyId input = (ColIdKeyId) obj;
+            return this.columnId == input.columnId && this.keyInfo.equals(input.keyInfo);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(columnId, keyInfo);
+        }
+    }
+
+    private static class ColIdSequenceId
+    {
+        private final int columnId;
+        private final int sequenceId;
+
+        public ColIdSequenceId(int inputColumnId, int inputSequenceId)
+        {
+            this.columnId = inputColumnId;
+            this.sequenceId = inputSequenceId;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColIdSequenceId)) {
```

Comment:
Presto doesn't use `id` in its orc language, it's column and sequence.

Reply:
I changed all the naming to be consistent with rest of the code in presto-orc

columnId -> column
sequenceId -> sequence 
keyInfo/keyId -> key